### PR TITLE
Edit and publish form workflow

### LINKED ISSUE #142: Form edit and publish workflow finalizing
As a user I'd like to be able to

- [x] - Go to edit latest draft version when clicking "edit" on production grade form
- [x] - Edit draft version in question when clicking edit on draft grade form
- [x] - Delete current draft version and create new one when clicking "save" on draft editor
- [x] - When clicking "publish"
- [x]    1. Archive current production grade form
- [x]    2. Change current draft to production grade
- [x]    3. Delete current draft

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #142

### CODE REVIEW

File: src/components/editor/editor-screen-table.tsx [hunk 1]
```diff
@@ -242,10 +266,20 @@ const EditorScreenTable: FC<Props> = ({
...
-      getActions: (row: GridRowParams) => [
-        renderActionCellEditItem(slug, row),
-        renderActionCellDeleteItem(row.id as string)
-      ]
+      getActions: (row: GridRowParams) => {
+        const versionRow = row.row as MetaformVersionRow;
+        if (versionRow.type === MetaformVersionType.Archived) {
+          return [
+            renderActionCellArchiveItem(row.id as string),
+            renderActionCellDeleteItem(row.id as string)
```
Comment (EVP1989) [inline src/components/editor/editor-screen-table.tsx:274]: Remove delete

File: src/components/screens/editor/editor-screen.tsx [hunk 2]
```diff
@@ -75,6 +75,81 @@ const EditorScreen: React.FC = () => {
...
+    const metaformToEdit = metaforms.find(metaform => metaform.id === id);
+
+    if (metaformToEdit) {
+      const version = metaformVersions.find(metaformVersion =>
+        (metaformVersion.data as Metaform).id === metaformToEdit.id &&
+        metaformVersion.type === MetaformVersionType.Draft);
+
+      if (version) {
+        navigate(`${currentPath}/${metaformToEdit.slug}/${version.id}`);
+        return;
```
Comment (EVP1989) [inline src/components/screens/editor/editor-screen.tsx:129]: Unnecessary return?

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: src/components/editor/editor-screen-table.tsx (hunk 1 above)
Comment (villejuutila) [reply src/components/editor/editor-screen-table.tsx:274]: Done

**Review by EVP1989 — APPROVED** [review]
Approved

**Pull request merged**